docs/rules/jsx-no-constructed-context-values.md

```diff
@@ -0,0 +1,37 @@
+# Prevent react contexts from taking non-stable values (react/jsx-no-constructed-context-values)
+
+This rule prevents non-stable values from being used as a value for `Context.Provider`.
+
+## Rule Details
+
+One way to resolve this issue may be to wrap the reference of the new value in a `useMemo()`. If it's a function then `useCallback()` can be used as well.
+
+If you _expect_ the context to be rerun on each render, then consider adding a comment/lint supression explaining why.
+
+## Examples
+
+❌ Examples of ***invalid*** code for this rule:
+
+```
+return (
+    <SomeContext.Provider value={{foo: 'bar'}}>
+        ...
+    </SomeContext.Provider>
+)
+```
+
+✅ Examples of ***valid*** code for this rule:
+
+```
+const foo = useMemo(() => {foo: 'bar'}, []);
+return (
+    <SomeContext.Provider value={foo}>
+        ...
+    </SomeContext.Provider>
+)
+```
+
+## Legitimate Uses
+React Context, and all its child nodes and Consumers are rerendered whenever the value prop changes. Because of Javascript's pass by reference nature, things like object expressions (`{foo: 'bar'}`) or function expressions get a new reference on every run through the component. This can cause needless rerenders and unintended consequences.
```

Comment:
JavaScript is strictly, alway, and only ever pass by value (see https://web.archive.org/web/20161005155047/http://www.jon-carlos.com/2013/is-javascript-call-by-value-or-call-by-reference/).

The issue is that like everything else in javascript, objects are compared by *identity*, not contents, and if you create a new object on every render, that identity changes across renders.

Reply:
My mistake, thanks for clarifying!